src/reducers/scenes/StakingReducer.ts

```diff
@@ -0,0 +1,148 @@
+import { combineReducers, Reducer } from 'redux'
+
+import { StakePlugin, StakePolicy, StakePosition } from '../../plugins/stake-plugins/types'
+import { Action } from '../../types/reduxTypes'
+
+interface AddStakePolicyAction {
+  type: 'STAKING/ADD_POLICY'
+  walletId: string
+  stakePolicy: StakePolicy
+}
+
+interface FinishAction {
+  type: 'STAKING/FINISH_LOADING'
+  walletId: string
+}
+
+interface UpdateLockedAmountAction {
+  type: 'STAKING/UPDATE_LOCKED_AMOUNT'
+  walletId: string
+  lockedNativeAmount: string
+}
+
+interface UpdatePluginsAction {
+  type: 'STAKING/UPDATE_PLUGINS'
+  walletId: string
+  stakePlugins: StakePlugin[]
+}
+
+interface UpdatePoliciesAction {
+  type: 'STAKING/UPDATE_POLICIES'
+  walletId: string
+  stakePolicies: StakePolicyMap
+}
+
+interface UpdatePositionAction {
+  type: 'STAKING/UPDATE_POSITION'
+  walletId: string
+  stakePolicyId: string
+  stakePosition: StakePositionMap[string]
+}
+
+interface UpdatePositionsAction {
+  type: 'STAKING/UPDATE_POSITIONS'
+  walletId: string
+  stakePositionMap: StakePositionMap
+}
+
+export type StakingAction =
+  | AddStakePolicyAction
+  | FinishAction
+  | UpdateLockedAmountAction
+  | UpdatePluginsAction
+  | UpdatePoliciesAction
+  | UpdatePositionAction
+  | UpdatePositionsAction
```

Comment:
I would prefer to put the action types inline, like this:

```ts
export type StakingAction =
| {
  type: 'STAKING/UPDATE_POSITIONS',
  ...
}
| { 
    type: 'STAKING/UPDATE_POLICIES'
    ...
}
```

This saves us from having to invent a bunch of names that aren't used anywhere else like `UpdatePositionsAction`, and therefore reduces the cognitive load / boilerplate.

Since we are stuck with Redux for the time being, our general goal has been to reduce as much boilerplate as much as possible to make life tolerable.